### ISSUE TITLE
Fl_Text_Display unable to scroll to bottom line when there's less than 1 full line to be scrolled

### ISSUE THREAD

**Issue by qartis** [fltk/fltk#1088]
**Describe the bug**
When the contents of an Fl_Text_Display are slightly taller than the widget's height, horizontal and vertical scrollbars appear but the view can't be scrolled down to expose the bottom line.

The bug will NOT happen if there are multiple overflowed lines that can be scrolled. The bug only happens when the widget's contents only overflow by a single line.

**To Reproduce**
```
#include <FL/Fl.H>
#include <FL/Fl_Window.H>
#include <FL/Fl_Text_Display.H>
int main() {
   Fl_Window *win = new Fl_Window(640, 480, "");
   Fl_Text_Display *disp = new Fl_Text_Display(20, 20, 640-40, 480-40);
   Fl_Text_Buffer *buff = new Fl_Text_Buffer();
   disp->buffer(buff);
   disp->textsize(50);

   # NOTE: no trailing newline
   buff->text("Line 1\nLine 2\nLine 3");
   win->resizable(*disp);
   win->show();
   return(Fl::run());
}
```

1. Resize the window so that the scrollbars appear and the horizontal scrollbar partially covers the final line
2. Note that the vertical scrollbar shows the view can be scrolled down but it's not possible to scroll down.

**Expected behavior**
The view should always allow the user to scroll down to fully expose the final line of text.

**Screenshots**
If Fl_Text_Display is resized so that only a single line overflows then the widget CANNOT be scrolled even though a vertical scrollbar is displayed. Screenshot:
![Screenshot 2024-10-10 19-36-29](https://github.com/user-attachments/assets/417630c0-dd95-4f28-9b6f-bbeb5278645c)

If the widget is made shorter so that 2 full lines overflow then scrolling is now possible. Note that because the widget is now slightly shorter than 2 full lines of text, the widget allows scrolling beyond the final line of text so that it's fully shown which is the expected behavior. Screenshots:
![Screenshot 2024-10-10 19-36-34](https://github.com/user-attachments/assets/ca4d00b0-ac0a-4096-a176-0cdb5f320977)
![Screenshot 2024-10-10 19-38-03](https://github.com/user-attachments/assets/e428f2e7-a83a-434d-a03a-f73bb2330fa8)
![Screenshot 2024-10-10 19-38-07](https://github.com/user-attachments/assets/ad829c1c-759f-4097-a439-4664200e2e7b)

**FLTK Version**
 - Version: 1.4.0 master at 013e93

**FLTK Configure / Build Options**
-- All drawing uses Cairo   Yes
-- Use Pango                   Yes

**Platform:**
 - OS: Linux Debian 12
 - X11
 - dwm

**Comment by MatthiasWM** [comment]
Thanks for the feedback. I did optimize scrolling a while ago. I will take a look at this in the next days.

**Comment by Albrecht-S** [comment]
@MatthiasWM AFAICT this is a long standing issue only visible if the text buffer doesn't end with a newline (`LF`) character. I can confirm the issue.

Possible related issue: problem if the Fl_Text_Display widget's height is smaller than 2 or 3 lines but I don't remember exactly what the issue was.

It would be "nice to have" if you could fix one or both of these issues - maybe it's the same root cause: line counting, but both issues should not block the release because they are benign.

**Comment by MatthiasWM** [comment]
I just pushed a fix to master. @qartis Let me know if this fixes the issue for you.

Note: the member variable `mNBufferLines` does not count trailing partial lines, so the sequence `"a\nb\nc"` sets `mNBufferLines` to 2 and `"a\nb\nc\n"` is 3. A method that reset the scrollbar was not aware of this. I decided to fix the method, but not the one-off of the variable to make sure that I don't cause side effects.
